sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/JobScheduleTerminatePollerAsync.java

```diff
@@ -0,0 +1,105 @@
+import com.azure.core.util.polling.PollResponse;
+import com.azure.core.util.polling.PollingContext;
+import reactor.core.publisher.Mono;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * Async poller class used by {@code beginTerminateJobSchedule} to implement polling logic for terminating a {@link BatchJobSchedule}.
+ * Returns {@link BatchJobSchedule} values during polling and as the final result.
+ */
+public final class JobScheduleTerminatePollerAsync {
+
+    private final BatchAsyncClient batchAsyncClient;
+    private final String jobScheduleId;
+    private final RequestOptions options;
+    private final Context requestContext;
+
+    /**
+     * Creates a new {@link JobScheduleTerminatePollerAsync}.
+     *
+     * @param batchAsyncClient The {@link BatchAsyncClient} used to interact with the Batch service.
+     * @param jobScheduleId The ID of the Job Schedule being terminated.
+     * @param options Optional request options for service calls.
+     */
+    public JobScheduleTerminatePollerAsync(BatchAsyncClient batchAsyncClient, String jobScheduleId,
+        RequestOptions options) {
+        this.batchAsyncClient = batchAsyncClient;
+        this.jobScheduleId = jobScheduleId;
+        this.options = options;
+        this.requestContext = options == null ? Context.NONE : options.getContext();
+    }
+
+    /**
+     * Activation operation to start the termination.
+     *
+     * @return A function that initiates the terminate request and returns a PollResponse with IN_PROGRESS status.
+     */
+    public Function<PollingContext<BatchJobSchedule>, Mono<PollResponse<BatchJobSchedule>>> getActivationOperation() {
+        return context -> batchAsyncClient.terminateJobScheduleWithResponse(jobScheduleId, options)
+            .thenReturn(new PollResponse<>(LongRunningOperationStatus.IN_PROGRESS, null));
+    }
+
+    /**
+     * Poll operation to check the job schedule's termination state.
+     *
+     * @return A function that polls the job schedule and returns a PollResponse with the current operation status.
+     */
+    public Function<PollingContext<BatchJobSchedule>, Mono<PollResponse<BatchJobSchedule>>> getPollOperation() {
+        return context -> {
+            RequestOptions pollOptions = new RequestOptions().setContext(this.requestContext);
+            return batchAsyncClient.getJobScheduleWithResponse(jobScheduleId, pollOptions).flatMap(response -> {
+                BatchJobSchedule jobSchedule = response.getValue().toObject(BatchJobSchedule.class);
+                BatchJobScheduleState state = jobSchedule.getState();
+
+                LongRunningOperationStatus status;
+                if (BatchJobScheduleState.TERMINATING.equals(state)) {
+                    status = LongRunningOperationStatus.IN_PROGRESS;
+                } else if (BatchJobScheduleState.COMPLETED.equals(state)) {
+                    status = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
```

Comment:
See the other comment about not asserting the final state. This is more likely to be correct than the job disable/enable case, but even here, I believe the job schedule could be in the 'Deleting' state for example, which doesn't mean a failure to terminate.